components/views/ProfilePage/schema.ts

```diff
@@ -4,8 +4,46 @@ export const formSchema = z.object({
   title: z.string().min(1).max(100),
   description: z.string().min(1).max(180),
   skills: z.string().min(2).max(160),
-  github: z.string().optional(),
-  linkedin: z.string().optional(),
-  twitter: z.string().optional(),
-  website: z.string().optional(),
+  github: z
+    .string()
+    .refine(
+      (value) =>
+        !value || new RegExp('^https://github.com/[a-zA-Z0-9_-]+$').test(value),
+      {
+        message: 'Invalid GitHub URL',
+      }
+    ),
+  linkedin: z
+    .string()
+    .refine(
+      (value) =>
+        !value ||
+        new RegExp('^(https?://)?(www.)?linkedin.com/in/[a-zA-Z0-9-]+/?$').test(
+          value
+        ),
+      {
+        message: 'Invalid LinkedIn profile URL',
+      }
+    ),
+  twitter: z
+    .string()
```

Comment:
Here also, we should mandatory `https` if user adds it.

Also www.twitter.com/rohit_dasu_ says wrong, it should allow www or https://

<img width="387" alt="Screenshot 2023-12-19 at 6 28 20 PM" src="https://github.com/rohitdasu/projectmate/assets/48400770/1bdd5e75-7705-41aa-81ee-9f0945e7ff38">
<img width="381" alt="Screenshot 2023-12-20 at 7 18 07 PM" src="https://github.com/rohitdasu/projectmate/assets/48400770/2a2b5855-0048-45ee-b786-e15136a8c7a7">